nodes/nomos-node/src/api/backend.rs

```diff
@@ -214,12 +233,108 @@ where
+    RuntimeServiceId: Debug
+        + Sync
+        + Send
+        + Display
+        + Clone
+        + 'static
+        + AsServiceId<
+            Cryptarchia<
+                Tx,
+                DaStorageSerializer,
+                SamplingBackend,
+                SamplingNetworkAdapter,
+                SamplingRng,
+                SamplingStorage,
+                DaVerifierBackend,
+                DaVerifierNetwork,
+                DaVerifierStorage,
+                TimeBackend,
+                ApiAdapter,
+                RuntimeServiceId,
+                SIZE,
+            >,
+        >
+        + AsServiceId<
+            DaVerifier<
+                DaShare,
+                DaVerifierNetwork,
+                DaVerifierBackend,
+                DaStorageSerializer,
+                RuntimeServiceId,
+            >,
+        >
+        + AsServiceId<
+            DaIndexer<
+                Tx,
+                DaBlobInfo,
+                DaVerifiedBlobInfo,
+                DaStorageSerializer,
+                SamplingBackend,
+                SamplingNetworkAdapter,
+                SamplingRng,
+                SamplingStorage,
+                DaVerifierBackend,
+                DaVerifierNetwork,
+                DaVerifierStorage,
+                TimeBackend,
+                ApiAdapter,
+                RuntimeServiceId,
+                SIZE,
+            >,
+        >
+        + AsServiceId<
+            nomos_da_network_service::NetworkService<
+                DaNetworkValidatorBackend<Membership>,
+                RuntimeServiceId,
+            >,
+        >
+        + AsServiceId<
+            nomos_network::NetworkService<
+                nomos_network::backends::libp2p::Libp2p,
+                RuntimeServiceId,
+            >,
+        >
+        + AsServiceId<StorageService<RocksBackend<DaStorageSerializer>, RuntimeServiceId>>
+        + AsServiceId<
+            TxMempoolService<
+                nomos_mempool::network::adapters::libp2p::Libp2pAdapter<
+                    Tx,
+                    <Tx as Transaction>::Hash,
+                    RuntimeServiceId,
+                >,
+                MockPool<HeaderId, Tx, <Tx as Transaction>::Hash>,
+                RuntimeServiceId,
+            >,
+        >
+        + AsServiceId<
+            DaMempoolService<
+                nomos_mempool::network::adapters::libp2p::Libp2pAdapter<
+                    DaVerifiedBlobInfo,
+                    DaVerifiedBlobInfo::BlobId,
+                    RuntimeServiceId,
+                >,
+                MockPool<HeaderId, DaVerifiedBlobInfo, DaVerifiedBlobInfo::BlobId>,
+                SamplingBackend,
+                SamplingNetworkAdapter,
+                SamplingRng,
+                SamplingStorage,
+                DaVerifierBackend,
+                DaVerifierNetwork,
+                DaVerifierStorage,
+                ApiAdapter,
+                RuntimeServiceId,
+            >,
```

Comment:
I'm wondering if this could be extracted somehow

Reply:
I mean, the only way would be a supertrait, which needs to anyway be defined somewhere. I don't think it makes things any better. Do you have any ideas? It is true that it will not scale well with more services, but I can't think of another way.

Reply:
Yeah, I do not think there is another way. In this case, I think we should probably migrate everything that has to deal with the node (outer layer) to use the exact types, as we have them defined in the crate already. That would simplify everything at a single point. Not to do in this PR ofc.

Reply:
Yes, the problem is that these are also used by the executor, so it won't compile anymore if we use concrete types. We would need to either duplicate code, or leave it as it is. As far as I can see.

Reply:
> Yes, the problem is that these are also used by the executor, so it won't compile anymore if we use concrete types. We would need to either duplicate code, or leave it as it is. As far as I can see.

That will not be a problem. We are gonna have to merge both in the same binary. So we will do it when doing that later on.

Reply:
Amazing 👍

Reply:
Then all these trait bounds will disappear.